src/main/service/listManipulation/magsStandardListService.ts

```diff
@@ -0,0 +1,90 @@
+import moment from 'moment-timezone';
+import {DataManipulationService} from '../dataManipulationService';
+import {DateTimeHelper} from '../../helpers/dateTimeHelper';
+
+const dataManipulationService = new DataManipulationService();
+const dateTimeHelper = new DateTimeHelper();
+
+export class MagsStandardListService {
+  public manipulatedMagsStandardListData(magsStandardListData: object, language: string, languageFile: string): object {
+    let hearingCount = 0;
+    magsStandardListData['courtLists'].forEach(courtList => {
+      courtList['courtHouse']['courtRoom'].forEach(courtRoom => {
+        courtRoom['session'].forEach(session => {
+          session['sittings'].forEach(sitting => {
+            const allHearings = [];
+            session['formattedJudiciaries'] = dataManipulationService.findAndManipulateJudiciary(sitting);
+            this.formatCaseTime(sitting, 'h:mma');
+            sitting['formattedDuration'] = dateTimeHelper.formatDuration(sitting['durationAsDays'] as number,
+              sitting['durationAsHours'] as number, sitting['durationAsMinutes'] as number, language, languageFile);
+            hearingCount = hearingCount + sitting['hearing'].length;
+            sitting['hearing'].forEach(hearing => {
+              if (hearing?.party) {
+                hearing.party.forEach(party => {
+                  const hearingString = JSON.stringify(hearing);
+                  const hearingObject = JSON.parse(hearingString);
+                  this.manipulatePartyInformation(hearingObject, party);
+                  hearingObject['case'].forEach(thisCase => {
+                    thisCase['formattedConvictionDate'] = dateTimeHelper.formatDate(thisCase['convictionDate'], 'DD/MM/YYYY');
+                    thisCase['formattedAdjournedDate'] = dateTimeHelper.formatDate(thisCase['adjournedDate'], 'DD/MM/YYYY');
+                  });
+                  allHearings.push(hearingObject);
+                });
+              }
+            });
+            sitting['hearing'] = allHearings;
+          });
+        });
+        courtRoom['totalHearing'] = hearingCount;
+        hearingCount = 0;
```

Comment:
Rather than setting hearingCount back to 0 could the let on line 10 be moved into one of the loops instead?

Reply:
We are not using hearingCount for this list. redundant code :) removed.